app/setup/executor.go

```diff
@@ -148,28 +161,46 @@ func (e *Executor) RegisterComponentSetupSteps() error {
 	return nil
 }
 
-func (e *Executor) createComponentSteps(namespace string) (componentSteps []ExecutorStep, waitSteps []ExecutorStep, err error) {
-	ecoSystemClient, err := componentEcoSystem.NewForConfig(e.ClusterConfig)
-	if err != nil {
-		return componentSteps, waitSteps, fmt.Errorf("failed to create K8s Component-EcoSystem client: %w", err)
+func (e *Executor) createLonghornSteps(componentsClient componentEcoSystem.ComponentInterface) []ExecutorStep {
+	var result []ExecutorStep
+	components := e.SetupContext.AppConfig.Components
+	namespace := e.SetupContext.AppConfig.TargetNamespace
+
+	if containsComponent(longhornComponentName, components) {
+		componentAttributes := components[longhornComponentName]
+		result = append(result, component.NewInstallComponentsStep(componentsClient, longhornComponentName, componentAttributes.HelmRepositoryNamespace, componentAttributes.Version, namespace, componentAttributes.DeployNamespace))
+		result = append(result, component.NewWaitForComponentStep(componentsClient, createComponentLabelSelector(longhornComponentName), namespace, component.DefaultComponentWaitTimeOut5Minutes))
+		delete(components, longhornComponentName)
```

Comment:
This is dense code which consists of several steps. I suggest a simple var extraction to enhance readbility. I understand that my version is longer but I have the feeling it increases the understandability.

Also the key-finding routine could be simplified so that `containsComponent()` can be deleted.

```suggestion
	_, containsLonghorn := components[longhornComponentName]
	
	if containsLonghorn {
		helmRepoNamespace := components[longhornComponentName].HelmRepositoryNamespace
		version := components[longhornComponentName].Version
		deployNamespace := components[longhornComponentName].DeployNamespace

		installStep := component.NewInstallComponentsStep(componentsClient, longhornComponentName, helmRepoNamespace, version, namespace, deployNamespace)
		selector := createComponentLabelSelector(longhornComponentName)
		waitStep := component.NewWaitForComponentStep(componentsClient, selector, namespace, component.DefaultComponentWaitTimeOut5Minutes)

		result = append(result, installStep)
		result = append(result, waitStep)
		delete(components, longhornComponentName)
```